ocp_resources/utils.py

```diff
@@ -17,15 +17,32 @@ def __str__(self):
 
 class TimeoutSampler:
     """
-    Samples the function output.
-
-    This is a generator object that at first yields the output of function
-    `func`. After the yield, it either raises instance of `TimeoutExpiredError` or
-    sleeps `sleep` seconds.
-
-    Yielding the output allows you to handle every value as you wish.
-
-    Feel free to set the instance variables.
+    Yields a given function's return value.
+
+    Before each following iteration (the next yield) over the TimeSampler object, the generator sleeps until it reaches
+    timeout and throws a TimeoutExpiredError exception.
```

Comment:
`the generator sleeps until it reaches
    timeout and throws a TimeoutExpiredError exception.`

this is confusing, seems like after the yield it will sleep till the timeout.

Reply:
corrected.